api/python/provisioner/commands/destroy_vm.py

```diff
@@ -0,0 +1,157 @@
+#
+# Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+from typing import Type
+import logging
+
+from .. import (
+    inputs,
+    errors,
+    config,
+    ALL_MINIONS
+)
+from ..utils import run_subprocess_cmd
+from ..salt import local_minion_id
+from ..pillar import (
+    PillarKey,
+    PillarResolver
+)
+from ..vendor import attr
+# TODO IMPROVE EOS-8473
+
+from .configure_setup import (
+    SetupType
+)
+from .deploy import (
+    build_deploy_run_args,
+    Deploy
+)
+
+
+logger = logging.getLogger(__name__)
+
+
+deploy_states = dict(
+    prvsnr=[
+        "system.storage.glusterfs.teardown.volume_remove",
+        "system.storage.glusterfs.teardown.remove_dir",
+        "system.storage.glusterfs.teardown.services",
+        "system.storage.glusterfs.teardown.package_remove"
+    ]
+)
+
+run_args_type = build_deploy_run_args(deploy_states)
```

Comment:
`stages` attr won't make any sense here since only teardown is considered. makes sense to inherit and re-define:
```
@attr.s(auto_attribs=True)
class XXX(run_args_type):
   stages: str = attr.ib(init=False, default=None)
```

Reply:
comment is addressed